egret/model_library/unit_commitment/params.py

```diff
@@ -230,18 +258,26 @@ def _warn_neg_impedence(m, v, l):
     model.HVDCLineOutOfService = Param(model.HVDCLines, model.TimePeriods, within=Boolean, default=False,
                                        initialize=TimeMapper(dc_branch_attrs.get('planned_outage', dict())))
 
-    _branch_penalties = dict()
-    _md_violation_penalties = branch_attrs.get('violation_penalty')
-    if _md_violation_penalties is not None:
-        for i, val in _md_violation_penalties.items():
+    _branch_penalties = {}
+    _branches_with_slack = []
+    for bn, branch in branches.items():
```

Comment:
I think we should have some sort of options or configuration object to control whether or nor things like slacks get added rather than inferring it from what is in the data dict. If someone wants to solve with slacks and then without, they have to modify the data dict. Perhaps more importantly, it looks like slacks will get added even if `slack_type == SlackType.NONE`.

Reply:
The existing behavior (in main) allows the user to add slacks to any transmission line by setting the `violation_penalty` to something other than `None`.

Perhaps `slack_type == SlackType.NONE` should override the `violation_penalties` set for the individual branches? And we leave them alone if the `slack_type` is anything else?

Reply:
I guess it should also be noted that in the case that somebody wants to solve with slacks and without: provided they don't set individual `violation_penalty` attributes on a branch, no update to the data dictionary is required. They can optionally set the `transmission_flow_violation_cost` system attribute to set the system-wide transmission violation cost.

Reply:
I think I would be fine with that. We could also raise an error if `slack_type == SlackType.NONE` and `violation_penalties` are specified.

Reply:
I opted to ignore the `violation_penalty` and issue a warning. That seems to be most common for data incongruities in Egret, probably because its my preference, and I think Egret chooses reasonable defaults which saves the user time if they don't otherwise need to modify their inputs.

Reply:
Sounds good. Thanks.